### DISCUSSION TITLE
rna_calc_inf.do_job in Jupyter Notebook: NameError: name 'args' is not defined

### DISCUSSION THREAD

**Discussion by mariagviegas** [mmagnus/rna-tools#112] [Q&A] [answered]
Dear mmagnus,

first I wanted to thank you for developing such a great package of tool for RNA analysis.

I am currently trying to calculate the INF of an unsolvated tRNA pdb with the 'rna_calc_in.do_job()' tool using Jupyter Notebook. However, when compiling my code, I get the following error: **NameError**: name 'args' is not defined. (I am attaching a screencapture of my code and the whole error message). Nevertheless, when checking the documentation, I do not know what args is or how I should define it. If you know, could you please tell me how to solve this error?

I am using Python 3.9.4, jupyter core  4.7.1 and jupyter-notebook  6.3.0.

Sorry for bothering you and thanks in advance for your help.

Best regards,
Maria Gonzalez Viegas

![mariagviegas_26 04 2021](https://user-images.githubusercontent.com/83213019/116063826-b1df6180-a685-11eb-84ef-3848fd30f68e.PNG)

**Comment by mmagnus** [comment, marked as answer]
Dear @mariagviegas, sorry for the late reply, please explain to me what you want to do. 

This function was not designed to be run directly from Jupyter this way, but anyway, to calculate INF you need two structures. INF is a comparison matric, that you can use to compare two networks of interactions and see how similar they are. 

Maybe what you want is a simple IN, interaction network. So you can also that with rna-tools, but if you need this only for one structure, maybe using a server will be easier http://rnapdbee.cs.put.poznan.pl (you can select multiple tools there for the job) or http://genesilico.pl/clarna/.

Let me know what you want to do and we will find the right way to do it :-)

**Reply by mariagviegas** [reply]
Dear @mmagnus, sorry I did not see your answer. 
At the end I decided to use a server, because the job was somehow urgent. 

Still, thanks a lot for your help!

**Reply by mmagnus** [reply]
Yeah, great, for small jobs I often use servers myself. 

Let me know if you need this running for more files, then I can show you how to do it :-)
